lib/profile/view_profile/sections/sections.dart

```diff
@@ -10,8 +10,13 @@ import './summary.dart';
  
 class ProfileSections extends StatelessWidget {
   final User user;
+  final String matchId;
+  final String id;
+  final bool isOwnProfile;
 
-  ProfileSections({Key key, @required this.user}) : super(key: key);
+  ProfileSections(
+    {Key key, @required this.user, this.id, this.matchId, this.isOwnProfile}
```

Comment:
What if instead of `user` and `isOwnProfile`, we had one argument that would take in a uid that belongs to that other user? Let's say we call it `otherUserId`.

If `otherUserId` is defined, we know that we are not viewing our own profile. If `otherUserId` is undefined, we can just make a call to get our current user's uid from the redux store. That way, we are not passing uid's down widgets when we can just fetch it in the specific widget we need it in. Unless I'm not understanding something?

Reply:
We're passing down `user` here because we're doing all the fetching (`getUser`) once in the parent widget which is `UserProfile`. Both of its children (`ProfileHeader` and `ProfileSections`) and its children's children (`ProfileContact`, `ProfileSummary`, etc.) need the user info so I don't think it would be efficient to be fetching from firebase that many times just for one page!

Also, from what I understand, @nayancprakash did all the ID comparison (`isOwnProfile = uid == profileId`) in the parent widget solely for conditional rendering in the `ProfileHeader` and `ProfileContact` and temporarily for the logout button.  Which I think is okay cus if he doesn't pass down `isOwnProfile`, he's gonna have to pass down `profileId` anyways and do the ID comparison within the widgets that need it. Either one is fine for me.

I guess my only suggestion here is that there's no need to pass down the `uid` from UserProfile to this widget then to `ProfileContact`.  You can just directly access the `uid` from the redux store in the widget where you need it since that's what global app state is for! 👍

Reply:
Also, if we plan to keep `isOwnProfile`, can we rename it to `viewOnly` or `readOnly`?